genai-perf/tests/test_file_input_retriever.py

```diff
@@ -37,193 +35,247 @@
 
 class TestFileInputRetriever:
 
-    def open_side_effects(self, filepath, *args, **kwargs):
-        queries_content = "\n".join(
-            [
-                '{"text": "What production company co-owned by Kevin Loader and Rodger Michell produced My Cousin Rachel?"}',
-                '{"text": "Who served as the 1st Vice President of Colombia under El Libertador?"}',
-                '{"text": "Are the Barton Mine and Hermiston-McCauley Mine located in The United States of America?"}',
-            ]
+    @staticmethod
+    def open_side_effect(filepath, *args, **kwargs):
+        single_prompt = '{"text": "What is the capital of France?"}\n'
+        multiple_prompts = (
+            '{"text": "What is the capital of France?"}\n'
+            '{"text": "Who wrote 1984?"}\n'
+            '{"text": "What is quantum computing?"}\n'
+        )
+        single_image = '{"image": "image1.png"}\n'
+        multiple_images = (
+            '{"image": "image1.png"}\n'
+            '{"image": "image2.png"}\n'
+            '{"image": "image3.png"}\n'
         )
-        passages_content = "\n".join(
-            [
-                '{"text": "Eric Anderson (sociologist) Eric Anderson (born January 18, 1968) is an American sociologist"}',
-                '{"text": "Kevin Loader is a British film and television producer. "}',
-                '{"text": "Barton Mine, also known as Net Lake Mine, is an abandoned surface and underground mine in Northeastern Ontario"}',
-            ]
+        multi_modal = (
+            '{"text": "What is this image?", "image": "image1.png"}\n'
+            '{"text": "Who is this person?", "image": "image2.png"}\n'
         )
+        deprecated_text_input = (
+            '{"text_input": "Who is Albert Einstein?"}\n'
+            '{"text_input": "What is the speed of light?"}\n'
+        )
+        conflicting_key = '{"text": "This is a conflicting key", "text_input": "This is a conflicting key"}\n'
 
         file_contents = {
-            "queries.jsonl": queries_content,
-            "passages.jsonl": passages_content,
+            "single_prompt.jsonl": single_prompt,
+            "multiple_prompts.jsonl": multiple_prompts,
+            "single_image.jsonl": single_image,
+            "multiple_images.jsonl": multiple_images,
+            "multi_modal.jsonl": multi_modal,
+            "deprecated_text_input.jsonl": deprecated_text_input,
+            "conflicting_key.jsonl": conflicting_key,
         }
-        return mock_open(
-            read_data=file_contents.get(filepath, file_contents["queries.jsonl"])
-        )()
-
-    mock_open_obj = mock_open()
-    mock_open_obj.side_effect = open_side_effects
+        filename = Path(filepath).name
+        return mock_open(read_data=file_contents.get(filename))()
 
+    @staticmethod
+    @patch("genai_perf.inputs.retrievers.file_input_retriever.FileInputRetriever._encode_image", return_value="mock_base64_image")
```

Comment:
Not super familiar with this use case, but why is this method both a static method and also patching things?

Reply:
It's static since it doesn't rely on an instance of the class. It's just a utility function for the tests. The patching is to mock the dependencies (returns a mock value instead of actually calling encode, to avoid making the tests a lot less readable with actual encoded values).